tools/integration-test/src/tests/channel_upgrade/manual_channel_upgrade.rs

```diff
@@ -104,24 +97,35 @@ impl BinaryChannelTest for ChannelUpgradeInitHandshake {
         .map_err(|e| eyre!("error creating height for timeout height: {e}"))?;
         let timeout = UpgradeTimeout::Height(timeout_height);
 
-        info!("Initialise channel upgrade process...");
+        info!("Set channel in (INITUPGRADE, OPEN) state...");
 
-        init_channel_upgrade(
-            channel,
+        channel.flipped().build_chan_upgrade_init_and_send(
```

Comment:
This is not related to the test, but shouldn't channel upgrade init be permissioned and require governance approval? If it is not permissioned, an attacker can potentially DOS a channel by repeatedly trying to upgrade the channel to a version that the counterparty chain do not support.

Reply:
A chain is able to cancel an upgrade if it doesn't agree on the proposal. Since only the TRY and ACK are "blocking", wouldn't this avoid the DOS scenario you described? https://github.com/cosmos/ibc/blob/main/spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md#channelstate

It also seems that it is left to the individual implementation to define the access-control, so I'm guessing when this will be deployed to chains it will be permissioned, https://github.com/cosmos/ibc/blob/main/spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md#upgrade-handshake

Reply:
Ok makes sense. But that also means the test here probably won't work if tested against chain binary that require governance approval. I guess we will update the tests when official chains start to adopt the protocol.